app/src/sharedTest/java/org/oppia/android/app/profile/PinPasswordActivityTest.kt

```diff
@@ -1214,6 +1214,160 @@ class PinPasswordActivityTest {
+    }
+  }
+
+  @Test
+  fun testPinPassword_withUser_inputPin_configChange_inputPinIsPersisted() {
+    ActivityScenario.launch<PinPasswordActivity>(
+      PinPasswordActivity.createPinPasswordActivityIntent(
+        context = context,
+        adminPin = adminPin,
+        profileId = userId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .perform(editTextInputAction.appendText("123"), closeSoftKeyboard())
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .check(matches(withText("123")))
+    }
+  }
+
+  @Test
+  fun testPinPassword_withAdmin_inputPin_exceedMaxLength_textIsTrimmed() {
+    ActivityScenario.launch<PinPasswordActivity>(
+      PinPasswordActivity.createPinPasswordActivityIntent(
+        context = context,
+        adminPin = adminPin,
+        profileId = adminId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .perform(editTextInputAction.appendText("1234567"), closeSoftKeyboard())
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .check(matches(withText("12345")))
+    }
+  }
+
+  @Test
+  fun testPinPassword_withUser_inputPin_exceedMaxLength_textIsTrimmed() {
+    ActivityScenario.launch<PinPasswordActivity>(
+      PinPasswordActivity.createPinPasswordActivityIntent(
+        context = context,
+        adminPin = adminPin,
+        profileId = userId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .perform(editTextInputAction.appendText("1234567"), closeSoftKeyboard())
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .check(matches(withText("123")))
+    }
+  }
+
+  @Test
+  fun testPinPassword_withAdmin__retainPinOnRotation_opensHomeActivity() {
+    TestPlatformParameterModule.forceEnableMultipleClassrooms(false)
+    ActivityScenario.launch<PinPasswordActivity>(
+      PinPasswordActivity.createPinPasswordActivityIntent(
+        context = context,
+        adminPin = adminPin,
+        profileId = adminId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .perform(editTextInputAction.appendText("123"), closeSoftKeyboard())
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.pin_password_input_pin_edit_text))
+        .perform(editTextInputAction.appendText("45"), closeSoftKeyboard())
+      testCoroutineDispatchers.runCurrent()
+      intended(hasComponent(HomeActivity::class.java.name))
+    }
+  }
+
+  @Test
+  fun testPinPassword_enablesClassroom_withAdmin_retainPinOnRotation_openClassroomListActivity() {
+    TestPlatformParameterModule.forceEnableMultipleClassrooms(true)
+    ActivityScenario.launch<PinPasswordActivity>(
```

Comment:
If you import `ActivityScenario` directly, you will avoid redudantly mentioning it in every test.
```suggestion
   launch<PinPasswordActivity>(
```

Reply:
Done.